labgrid/driver/rawnetworkinterfacedriver.py

```diff
@@ -69,20 +70,21 @@ def start_record(self, filename, *, count=None):
         cmd = ["tcpdump", self.iface.ifname]
         if count is not None:
             cmd.append(str(count))
+        if timeout is not None:
+            cmd.append("--duration")
+            cmd.append(str(timeout))
         cmd = self._wrap_command(cmd)
-        with open(filename, "wb") as outdata:
-            self._record_handle = subprocess.Popen(cmd, stdout=outdata, stderr=subprocess.PIPE)
+        if filename == "-":
+            self._record_handle = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.DEVNULL)
+        else:
+            with open(filename, "wb") as outdata:
+                self._record_handle = subprocess.Popen(cmd, stdout=outdata, stderr=subprocess.PIPE)
         return self._record_handle
 
     @Driver.check_active
-    @step(args=["timeout"])
     def stop_record(self, *, timeout=None):
```

Comment:
Why do you keep timeout here? For backwards compatibility? I'm unsure whether we should do that.